src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -108,8 +163,43 @@ class TasksController(
     case _ => None
   }
 
+  @SuppressWarnings(Array("all")) // async/await
+  private def doKillTasks(toKillFuture: Future[Map[PathId, Seq[Instance]]], tasksIdToAppId: Map[Id, PathId], wipe: Boolean)(implicit identity: Identity): Future[Seq[EnrichedTask]] = async {
+    val toKill = await(toKillFuture)
+    val affectedApps = tasksIdToAppId.values.flatMap(appId => groupManager.app(appId))(collection.breakOut)
+
+    val killedTasks = await(Future.sequence(toKill
+      .filter { case (appId, _) => affectedApps.exists(app => app.id == appId) }
```

Comment:
Why do we have to filter here again? It's a little hard to follow how we come up with the right tasks to kill. So `toKill` gives a map of app ids (`PathId`) to a sequence of instances. Why do we require `tasksIdToAppId` then?

Reply:
I tried to refactor it but it was still not enough (I agree). I will try to dive into this deeper once more

Reply:
Sooo ... I found out this filter was introduced by @unterstein so I will go over the code with him next week. But I think it is not blocker of merging this patch because this code is there even in the taskResource so it should be working...

Reply:
Ok, let's not touch it in this diff then :).